script/check-simulation-build-autotools

```diff
@@ -0,0 +1,185 @@
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+set -euxo pipefail
+
+readonly OT_BUILD_JOBS=$(getconf _NPROCESSORS_ONLN)
+
+reset_source()
+{
+    git checkout -- .
+    git clean -xfd
+}
+
+build_default()
+{
+    reset_source
+
+    ./bootstrap
+    make -f examples/Makefile-simulation
+}
+
+build_all_features()
+{
+    local options=(
+        "-DOPENTHREAD_CONFIG_ANNOUNCE_SENDER_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_BORDER_AGENT_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_CHANNEL_MANAGER_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_CHILD_SUPERVISION_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_COAP_API_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_COAP_SECURE_API_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_COMMISSIONER_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_DHCP6_CLIENT_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_DHCP6_SERVER_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_DIAG_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_DNS_CLIENT_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_ECDSA_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_HEAP_EXTERNAL_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_IP6_FRAGMENTATION_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_IP6_SLAAC_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_LEGACY_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_MAC_BEACON_RSP_WHEN_JOINABLE_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_MAC_RETRY_SUCCESS_HISTOGRAM_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_MLE_ATTACH_BACKOFF_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_MLE_STEERING_DATA_SET_OOB_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_MPL_DYNAMIC_INTERVAL_ENABLE"
+        "-DOPENTHREAD_CONFIG_JAM_DETECTION_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_JOINER_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_LINK_RAW_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_LOG_LEVEL_DYNAMIC_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_MAC_FILTER_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_NCP_UART_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_PLATFORM_FLASH_API_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_PLATFORM_USEC_TIMER_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_SNTP_CLIENT_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_SOFTWARE_ACK_TIMEOUT_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_SOFTWARE_CSMA_BACKOFF_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_SOFTWARE_ENERGY_SCAN_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_SOFTWARE_RETRANSMIT_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_TMF_NETWORK_DIAG_MTD_ENABLE=1"
+        "-DOPENTHREAD_CONFIG_UDP_FORWARD_ENABLE=1"
+    )
+
+    reset_source
+    export CPPFLAGS="${options[*]} -DOPENTHREAD_CONFIG_LOG_LEVEL=OT_LOG_LEVEL_NONE"
+    ./bootstrap
```

Comment:
bootstrap costs a lot of time. How about just removing `build` and `output` and `tmp` in reset_source?